service/ports/rpc/handler_blobs_get.go

```diff
@@ -1,35 +1,75 @@
 package rpc
 
 import (
+	"bytes"
 	"context"
 	"io"
 
 	"github.com/boreq/errors"
+	"github.com/planetary-social/go-ssb/service/app/queries"
 	"github.com/planetary-social/go-ssb/service/domain/messages"
 	"github.com/planetary-social/go-ssb/service/domain/transport/rpc"
 	"github.com/planetary-social/go-ssb/service/domain/transport/rpc/mux"
 )
 
-type BlobStorage interface {
-	Get() (io.ReadCloser, error)
+const (
+	// MaxBlobChunkSizeInBytes specifies the max size of one response payload
+	// when sending blobs. Larger blobs are broken into several payloads. This
+	// is done to give other pieces of code some wire time and avoid blocking
+	// the connection completely when sending the blob.
+	MaxBlobChunkSizeInBytes = 100 * 1000
+)
+
+type GetBlobQueryHandler interface {
+	Handle(query queries.GetBlob) (io.ReadCloser, error)
 }
 
 type HandlerBlobsGet struct {
+	handler GetBlobQueryHandler
 }
 
-func NewHandlerBlobsGet() *HandlerBlobsGet {
-	return &HandlerBlobsGet{}
+func NewHandlerBlobsGet(handler GetBlobQueryHandler) *HandlerBlobsGet {
+	return &HandlerBlobsGet{
+		handler: handler,
+	}
 }
 
 func (h HandlerBlobsGet) Procedure() rpc.Procedure {
 	return messages.BlobsGetProcedure
 }
 
 func (h HandlerBlobsGet) Handle(ctx context.Context, w mux.ResponseWriter, req *rpc.Request) error {
-	_, err := messages.NewBlobsGetArgumentsFromBytes(req.Arguments())
+	args, err := messages.NewBlobsGetArgumentsFromBytes(req.Arguments())
 	if err != nil {
 		return errors.Wrap(err, "invalid arguments")
 	}
 
-	return errors.New("not implemented")
+	query := queries.GetBlob{
+		Id: args.Id(),
+	}
+
+	rc, err := h.handler.Handle(query)
+	if err != nil {
+		return errors.Wrap(err, "error executing the query")
+	}
+	defer rc.Close()
+
+	buf := &bytes.Buffer{}
+
+	for {
+		n, err := io.Copy(buf, io.LimitReader(rc, MaxBlobChunkSizeInBytes))
+		if err != nil {
+			return errors.Wrap(err, "failed to copy into buffer")
+		}
+
+		if n == 0 {
+			return nil
```

Comment:
I noticed that we may be missing a proper stream closing mechanism in `Mux`: https://github.com/planetary-social/go-ssb/blob/0a213f9c9316ed2633e14e19149c0e6a2534eba6/service/domain/transport/rpc/mux/mux.go#L47

We only close a stream once an error is returned, but we don't close it on `nil` while I think we should do so as well in this case according to the spec. 

If that's the case, please make sure it's covered at least in unit tests of the `Mux`.

Reply:
Interesting, I was under the impression that I had tests for this. I will create a ticket for this and solve this separately if you don't mind.

Reply:
It is possible that this happens through some kind of a convoluted side-effect btw.

Reply:
https://github.com/planetary-social/go-ssb/issues/11